odoo/addons/test_lint/tests/lint_case.py

```diff
@@ -16,7 +16,23 @@ def iter_module_files(self, *globs):
         """
         for modroot in map(get_module_path, get_modules()):
             for root, _, fnames in os.walk(modroot):
-                fnames = [j(root, n) for n in fnames]
+                fnames = [os.path.join(root, n) for n in fnames]
                 for glob in globs:
                     fnames = fnmatch.filter(fnames, glob)
                 yield from fnames
+
+
+class NodeVisitor(ast.NodeVisitor):
+    def visit(self, node):
+        method = 'visit_' + node.__class__.__name__
+        visitor = getattr(self, method, self.generic_visit)
+        return visitor(node)
+
+    def generic_visit(self, node):
+        for _field, value in ast.iter_fields(node):
```

Comment:
Wouldn't `ast.iter_child_node` better do the job? IIRC as its name implies it only returns AST nodes, and automatically deref's lists.

Reply:
Indeed https://github.com/python/cpython/blob/421b543311c2eee67482bba894e475a2696423d5/Lib/ast.py#L256